vSMR/SMRPlugin.cpp

```diff
@@ -722,10 +725,13 @@ void CSMRPlugin::OnTimer(int Counter)
 	// Retry login
 	if (!HoppieConnected && nextLoginRetryTime > 0 && time(NULL) >= nextLoginRetryTime)
 	{
+		isConnecting = true;
 		_beginthread(datalinkLogin, 0, NULL);
 		nextLoginRetryTime = 0; // Prevent multiple triggers
 	}
 
+	
+	//Auto connection
 	if (AutoConnect && !HoppieConnected && !isConnecting)

```

Comment:
I don't believe this is properly resolved. The effect now is that, if auto-connect is enabled, repeated connection attempts will be made (without cooldown!) whenever Hoppie's is not connected and a connection attempt is not currently in progress. This also means that, as before, the user is unable to disconnect themselves from Hoppie's if auto-connect is enabled. It also appears the checks which ensure the user is actually connected and controlling have been lost from the last PR.

Reply:
Thank you for mentioning that I will re-add them in also learning for an exam at the same time, just coding between breaks